opds.py

```diff
@@ -1409,43 +1414,98 @@ def show_current_entrypoint(self, entrypoint):
     def add_breadcrumbs(self, lane, include_lane=False, entrypoint=None):
         """Add list of ancestor links in a breadcrumbs element.
 
+        :param lane: Add breadcrumbs from up to this lane.
+        :param include_lane: Include `lane` itself in the breadcrumbs.
+        :param entrypoint: The currently selected entrypoint, if any.
+
         TODO: The switchover from "no entry point" to "entry point" needs
         its own breadcrumb link.
         """
-        # Ensure that lane isn't top-level before proceeding
+        if entrypoint is None:
+            entrypoint_query = ""
+        else:
+            entrypoint_query = "?entrypoint=" + entrypoint.INTERNAL_NAME
+
+        # Breadcrumbs for lanes may be end up being cut off by a
+        # patron-type-specific root lane. If so, that lane -- not the
```

Comment:
What is a patron type in this situation? It's also in the description but I'm not sure what it refers to.

Reply:
It's the value of `Patron.external_type`.